### PR TITLE
Add smoke test for go and haskell initializers (and fix the go one)

### LINKED ISSUE #333: Add smoke test for go and haskell initializers

### PR DESCRIPTION
**Pull request by soenkehahn**
Fixes #333.

### CODE REVIEW

File: test/spec/TestUtils.hs [hunk 1]
```diff
@@ -186,3 +188,44 @@ onTestFailureLogger test = do
...
+      when (not running) $ do
+        threadDelay 100000
+        waitUntilRunning
+
+    isRunning = do
+      Exit c <-
+        cmd
+          "curl --silent localhost:8777/base.ts"
+          (EchoStdout False)
+      pure $ c == ExitSuccess
```
Comment (soenkehahn) [inline test/spec/TestUtils.hs:231]: This just moved here unchanged from `ExampleSpec`.

File: ts/go/mod.ts [hunk 2]
```diff
@@ -10,6 +10,8 @@ import { GOMOD2NIX_REPO } from "./consts.ts";
 const getGoModNixToml = (src: string): NixExpression => {
   const gen = new Deno.Command("nix", {
     args: [
+      "--extra-experimental-features",
+      "nix-command flakes",
```
Comment (soenkehahn) [inline ts/go/mod.ts:14]: This is a fix for the go initializer that the test revealed.

File: test/spec/InitSpec.hs [hunk 3]
```diff
@@ -4,90 +4,144 @@ module InitSpec where
 
 import Data.String.Interpolate (i)
 import Data.String.Interpolate.Util (unindent)
+import Development.Shake (StdoutTrim (..), cmd, cmd_)
```
Comment (soenkehahn) [inline test/spec/InitSpec.hs:7]: This file is probably best reviewed with white space changes switched off.

File: test/spec/ExampleSpec.hs [hunk 4]
```diff
@@ -51,28 +30,6 @@ retryGet url = fmap cs <$> go 1000
...
-        threadDelay 100000
-        waitUntilRunning
-
-    isRunning = do
-      Exit c <-
-        cmd
-          "curl --silent localhost:8777/base.ts"
-          (EchoStdout False)
-      pure $ c == ExitSuccess
-
```
Comment (soenkehahn) [inline test/spec/ExampleSpec.hs:75]: All this just moved to `TestUtils` unchanged.

File: test/spec/InitSpec.hs [hunk 5]
```diff
@@ -4,90 +4,144 @@ module InitSpec where
...
+            }
+          |]
+        output <- runGarn ["init"] "" repoDir Nothing
+        onTestFailureLog output
+        stderr output `shouldBe` "[garn] Creating a garn.ts file\n"
+        let check = do
+              output <- runGarn ["run", "myProject.start"] "" repoDir Nothing
+              onTestFailureLog output
+              stdout output `shouldContain` "starting my project..."
+        check
```
Comment (alexdavid) [inline test/spec/InitSpec.hs:131]: I think this check will cause the problem of failing anytime we introduce a breaking change to the IPC (which is what I ran into on #336). I'm wondering if we should just remove these non-replaced checks

**Review by alexdavid — APPROVED** [review]
This looks good, but I think we need to remove the double check now that I think about it

**Review by alexdavid — APPROVED** [review]

**Pull request merged**